code/RealMeService.php

```diff
@@ -320,205 +465,291 @@ public function getBackURL()
     }
 
-    /**
-     * @return string|null Either the directory where temp files can be written by SimpleSAMLphp, or null if undefined
-     */
-    public function getTempDir()
+    public function getSPCertContent($contentType = 'certificate')
     {
-        return (defined('REALME_TEMP_DIR') ? REALME_TEMP_DIR : null);
+        return $this->getCertificateContents($this->getSigningCertPath(), $contentType);
+    }
+
+    public function getIdPCertContent()
+    {
+        return $this->getCertificateContents(Controller::join_links($this->getIdPCertPath()), 'certificate');
     }
 
     /**
-     * This looks first to a Config variable that can be set in YML configuration, and falls back to generating a
-     * salted SHA256-hashed password. To generate a password in this format, see the bin/pwgen.php file in the
-     * SimpleSAMLphp vendor directory (normally vendor/madmatt/simplesamlphp/bin/pwgen.php). If setting a password
-     * via Config, ensure it contains {SSHA256} at the start of the line.
+     * Returns the content of the SAML signing certificate. This is used by getAuth() and by RealMeSetupTask to produce
+     * metadata XML files.
      *
-     * @return string|null The administrator password set for SimpleSAMLphp. If null, it means a strong hash couldn't be
-     * created due to the code being deployed on an older machine, and a generated password will need to be set.
+     * @param string $certPath The filesystem path to where the certificate is stored on the filesystem
+     * @param string $contentType Either 'certificate' or 'key', depending on which part of the file to return
+     * @return string|null The content of the signing certificate
      */
-    public function findOrMakeSimpleSAMLPassword()
+    public function getCertificateContents($certPath, $contentType = 'certificate')
     {
-        if (strlen($this->config()->simplesaml_hashed_admin_password) > 0) {
-            $password = $this->config()->simplesaml_hashed_admin_password;
+        $text = null;
 
-            if (strpos($password, '{SSHA256}') !== 0) {
-                $password = null; // Ensure password is salted SHA256
-            }
-        } else {
-            $salt = openssl_random_pseudo_bytes(8, $strongSalt); // SHA256 needs 8 bytes
-            $password = openssl_random_pseudo_bytes(32, $strongPassword); // Make a random 32-byte password
+        if (!is_null($certPath)) {
+            $certificateContents = file_get_contents($certPath);
 
-            if (!$strongSalt || !$strongPassword || !$salt || !$password) {
-                $password = null; // Ensure the password is strong, return null if we can't guarantee a strong one
+            // If the file does not contain any header information and the content type is certificate, just return it
+            if($contentType == 'certificate' && !preg_match('/-----BEGIN/', $certificateContents)) {
```

Comment:
@dhensby To help us get this pull request through, it might be worthwhile retro-actively raising this as an issue with a dedicated pull request. Would you be ok with that?